Quadriga/src/main/java/edu/asu/spring/quadriga/exceptions/QuadrigaExceptionHandler.java

```diff
@@ -67,4 +67,12 @@ public ModelAndView handle404Exception(Quadriga404Exception ex) {
         return modelAndView;
     }
 
+    @ExceptionHandler(value = { IllegalObjectException.class })
+    public ModelAndView handleInvalidCastException(IllegalObjectException ice) {
+        ModelAndView modelAndView = new ModelAndView("auth/404");
```

Comment:
if there doesn't exist one already. But I assume we have a generic error page for cases like this.